ReSharper.FSharp/src/FSharp.Psi/src/Impl/DeclaredElement/FSharpPropertyAccessor.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using FSharp.Compiler.SourceCodeServices;
+using JetBrains.Annotations;
+using JetBrains.Metadata.Reader.API;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Util;
+using JetBrains.ReSharper.Plugins.FSharp.Util;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.ExtensionsAPI;
+using JetBrains.ReSharper.Psi.Impl;
+using JetBrains.ReSharper.Psi.Modules;
+using JetBrains.ReSharper.Psi.Resolve;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.Util;
+using JetBrains.Util.DataStructures;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.DeclaredElement
+{
+  public class FSharpPropertyAccessor : IAccessor
```

Comment:
This class seems to reimplement too many things. Why is it better than reusing the available base ones?